proposals/params-collections.md

```diff
@@ -0,0 +1,171 @@
+# `params Collections`
+
+## Summary
+
+In C# 12 language added support for creating instances of collection types beyond just arrays.
+See [collection expressions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md).
+This proposal extends `params` support to all such collection types.
+
+## Motivation
+
+A `params` array parameter provides a convenient way to call a method that takes an arbitrary length list of arguments.
+Today `params` parameter must be an array type. However, it might be beneficial for a developer to be able to have the same
+convenience when calling APIs that take other collection types. For example, an `ImmutableArray<T>`, `ReadOnlySpan<T>`, or 
+plain `IEnumerable`. Especially in cases when compiler is able to avoid an implicit array allocation for the purpose of
+creating the collection (`ImmutableArray<T>`, `ReadOnlySpan<T>`, etc). Today, in situations when an API takes a collection type,
+developers add `params` overload that takes an array, construct the target collection and call the original overload with that
+collection, thus consumers of the API have to trade an extra array allocation for convenience.
+
+## Detailed design
+
+### Method parameters
+
+The [Method parameters](https://github.com/dotnet/csharpstandard/blob/draft-v9/standard/classes.md#1562-method-parameters) section is adjusted as follows.
+
+```diff ANTLR
+formal_parameter_list
+    : fixed_parameters
+-    | fixed_parameters ',' parameter_array
++    | fixed_parameters ',' parameter_collection
+-    | parameter_array
++    | parameter_collection
+    ;
+
+-parameter_array
++parameter_collection
+-    : attributes? 'params' array_type identifier
++    : attributes? 'params' 'scoped'? type identifier
+    ;
+```
+
+A *parameter_collection* consists of an optional set of *attributes*, a `params` modifier, an optional `scoped` modifier,
+a *type*, and an *identifier*. A parameter collection declares a single parameter of the given type with the given name.
+The *type* of a parameter collection shall be one of the following valid target types for a collection expression
```

Comment:
Could we not simplify this list by just specifying that there must exist a `collection expression conversion` from a collection expression to the target type?

Reply:
> Could we not simplify this list by just specifying that there must exist a collection expression conversion from a collection expression to the target type?

First, there is no collection expression. Second, I intentionally exclude the nullable value type case.